apollo-router/src/plugins/telemetry/metrics/filter.rs

```diff
@@ -0,0 +1,302 @@
+use std::sync::Arc;
+
+use buildstructor::buildstructor;
+use opentelemetry::metrics::noop::NoopMeterProvider;
+use opentelemetry::metrics::Counter;
+use opentelemetry::metrics::Histogram;
+use opentelemetry::metrics::InstrumentProvider;
+use opentelemetry::metrics::Meter;
+use opentelemetry::metrics::MeterProvider;
+use opentelemetry::metrics::ObservableCounter;
+use opentelemetry::metrics::ObservableGauge;
+use opentelemetry::metrics::ObservableUpDownCounter;
+use opentelemetry::metrics::Unit;
+use opentelemetry::metrics::UpDownCounter;
+use opentelemetry::Context;
+use opentelemetry::InstrumentationLibrary;
+use regex::Regex;
+
+pub(crate) struct FilterMeterProvider<T: MeterProvider> {
+    delegate: T,
+    deny: Option<Regex>,
+    allow: Option<Regex>,
+}
+
+#[buildstructor]
+impl<T: MeterProvider> FilterMeterProvider<T> {
+    #[builder]
+    fn new(delegate: T, deny: Option<Regex>, allow: Option<Regex>) -> Self {
+        FilterMeterProvider {
+            delegate,
+            deny,
+            allow,
+        }
+    }
+
+    pub(crate) fn apollo_metrics(delegate: T) -> Self {
+        FilterMeterProvider::builder()
+            .delegate(delegate)
+            .allow(
+                Regex::new(r"apollo\.router\.(operations?|config)(\..*|$)")
+                    .expect("regex should have been valid"),
+            )
+            .build()
+    }
+
+    pub(crate) fn public_metrics(delegate: T) -> Self {
+        FilterMeterProvider::builder()
+            .delegate(delegate)
+            .deny(
+                Regex::new(r"apollo\.router\.(operations?|config)(\..*|$)")
+                    .expect("regex should have been valid"),
+            )
+            .allow(
+                Regex::new(r"apollo\.router\.operations\.authentication")
+                    .expect("regex should have been valid"),
+            )
```

Comment:
what's the rationale on what is sent to other exporters and what isn't? Example: I followed the new conventions for https://github.com/apollographql/router/commit/dd7bac0e9269fbe8c1b3be431cf9e4373862d084, but that new metric was first thought for usrs, to for analytics, so do I need to add an exemption here?